factory/Cargo.toml

```
@@ -0,0 +1,20 @@
+[package]
+name = "factory"
+version = "0.1.0"
+edition = "2018"
```

Let's use the latest edition.
```suggestion
edition = "2021"
```